libs/wingc/src/type_check.rs

```diff
@@ -1740,25 +1825,50 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
-	fn add_module_to_env(&mut self, env: &mut SymbolEnv, module_name: String, statement_idx: usize) {
-		// TODO Hack: treat "cloud" or "std" as "_ in wingsdk" until I figure out the path issue
-		if module_name == "cloud" || module_name == "fs" || module_name == WINGSDK_STD_MODULE {
-			let mut wingii_types = wingii::type_system::TypeSystem::new();
+	fn add_module_to_env(
+		&mut self,
+		env: &mut SymbolEnv,
+		library_name: String,
+		namespace_filter: Vec<String>,
+		identifier: &Symbol,
+		statement_idx: usize,
+	) {
+		let mut wingii_types = wingii::type_system::TypeSystem::new();
+
+		// Loading the SDK is handled different from loading any other jsii modules because with the SDK we provide an exact
+		// location to locate the SDK, whereas for the other modules we need to search for them from the source directory.
+		let assembly_name = if library_name == WINGSDK_ASSEMBLY_NAME {
+			// in runtime, if "WINGSDK_MANIFEST_ROOT" env var is set, read it. otherwise set to "../wingsdk" for dev
```

Comment:
> otherwise set to "../wingsdk" for dev

This smells like a potential security issue. Usually dev behavior is something you want to be explicit, not implicit.

(but that's for a future fix, maybe open an issue).

Reply:
https://github.com/winglang/wing/issues/1518